### ISSUE TITLE
no procedures to add a coin / algorithum

### ISSUE THREAD

**Issue by keefer223** [oliverw/miningcore#1412]
Though there are supposedly 130 coins that can be used with miningcore and appear to be the popular ones. It appears ClusterConfig.cs is focused on only 5 algorithms / families: bitcoin, equihash, cryptonote, etherum, and ergo. Which are a good starting point, but there are more algorithms out there beside those. There is also NO documentation on how to add both algorithms and coins to miningcore.

I did do a search on the net for just that and the closest I could find were references to some old issues that focus on Issue #61 as a reference, I'm sorry to say there appears to have been some coding changes over the last 5 years. #61 references 
modifying: ClusterConfig.cs, CoinMataData.cs (not found),  BitcoinPayoutHandler.cs, BitcoinPool.cs (not found), and  BitcoinProperties.cs (not found).

With new coins coming out weekly, plus the 4,000 plus already out-there, one can not expect you to add all of them to miningcore. Thus a good set of procedures would allow the pool operators to expand the coins being offered to mine.

**Expected behavior**
one would expect a admin that would perpetuate the addition of both algorithm and coins.

And no I don't have a specific coin/ 's that I'd like to see added. My focus is on non asci coins and new coins, in order to get on the ground floor before they get major attention. Now, miningcore appears to have the most popular coins.

**Comment by oliverw** [comment]
I'm at a bit of a loss here. Adding coins can be extremely easy or very hard. It depends but you always need a solid understanding of how MC works.

**Comment by keefer223** [comment]
In either case, easy or hard, don't you think there should be
documentation? I know I'm not the only one looking for some documentation.
even the documentation for api is for the advanced programmer. This isn't
to trash the system but to open the possibilities to other coins. If you'd
like to outline how to do it so the average programmer / operator can
utilize it which could lead to a backend template. I'd be more than happy
to start a documentation.

thanks.

**Comment by oliverw** [comment]
I agree, but keep in mind that no amount of documentation will fully prepare you for the task. Coins come with so many intricacies and special cases which you only discover after digging into the codebase of the coin - if ever. God forbid that its an entirely new family that is not yet covered in MC. That case is a ton of work (usually takes me a full months of work).

**Comment by seperatis** [comment]
@keefer223 Sorry friend, but it's not @oliverw's job **to teach you stuff you don't know**. This is open-source software, you're free to tear it apart and learn how it works. Having the "expectation" that it's someone else's **responsibility** to show you how things work comes off as a bit _entitled._ 

I've thought about improving the documentation myself, but I haven't had time to do it. But... that's how Open-Source, (FREE), software works. Find something that needs fixed, and fix it. 

>even the documentation for api is for the advanced programmer.
What? This is categorical BS - I'm not an advanced programmer and I was able to build an entire website/front-end from scratch using the API, and setup a Cluster Configuration with global stratum servers in like 4 weeks.

How? I **read through old Issues** and I _politely_ asked questions when I would get stuck.

It's not @oliverw's or anyone else's job to teach you how things work in Miningcore. He's made this awesome software available free for anyone to use, improve, or make their own private repo with. He's not **also** somehow obligated to fullfil everyone's desires and wishes for features, and teach classes on how the code works line-by-line.

**This is not a bug.**

Closing.
